src/staging/staging_main.py

```diff
@@ -87,36 +91,86 @@ def run_staging(
     check_file_exists(snapshot_path)
 
     # load and parse the snapshot data json file
-    snapdata = load_json(snapshot_path)
-    contributors_df, responses_df = spp_parser.parse_snap_data(snapdata)
-
-    # the anonymised snapshot data we use in the DevTest environment
-    # does not include data in several columns. This fix should be removed
-    # when new anonymised data is given.
-    if network_or_hdfs == "hdfs" and config["global"]["dev_test"]:
-        responses_df["instance"] = 0
-        # In the anonymised data the selectiontype is always 'L'
-        col_size = contributors_df.shape[0]
-        random.seed(seed=42)
-        contributors_df["selectiontype"] = random.choice(["P", "C", "L"], size=col_size)
-        cellno_list = config["devtest"]["seltype_list"]
-        contributors_df["cellnumber"] = random.choice(cellno_list, size=col_size)
-    StagingMainLogger.info("Finished Data Ingest...")
-
-    val.validate_data_with_schema(contributors_df, "./config/contributors_schema.toml")
-    val.validate_data_with_schema(responses_df, "./config/long_response.toml")
-
-    # Data Transmutation
-    StagingMainLogger.info("Starting Data Transmutation...")
-    full_responses = processing.full_responses(contributors_df, responses_df)
-
-    # Validate and force data types for the full responses df
-    # TODO Find a fix for the datatype casting before uncommenting
-    val.combine_schemas_validate_full_df(
-        full_responses,
-        "./config/contributors_schema.toml",
-        "./config/wide_responses.toml",
-    )
+    # Check if feather file exists in snapshot path
+    feather_path = paths["feather_path"]
+    load_from_feather = config["global"]["load_from_feather"]
+    feather_files = [f for f in os.listdir(feather_path) if f.endswith(".feather")]
```

Comment:
I've tried running in DevTest and this line fails if there are no existing .feather files.

Exception has occurred: FileNotFoundError
[Errno 2] No such file or directory: '/ons/rdbe_dev/staging/feather'
  File "/home/cdsw/research-and-development/src/staging/staging_main.py", line 97, in run_staging